app/models/iapStore/iap-store.ts

```diff
@@ -0,0 +1,98 @@
+import RNIap, {
+  Product,
+  Purchase,
+  PurchaseError,
+  purchaseErrorListener,
+  purchaseUpdatedListener,
+} from "react-native-iap";
+
+import { Instance, SnapshotOut, flow, types } from "mobx-state-tree"
+import { observable } from "mobx"
+
+const SKU_COM_OICE_MEMBERSHIP = "com.oice.membership"
+
+/**
+ * Store IAP related information.
+ */
+export const IAPStoreModel = types
+  .model("IAPStore")
+  .extend(self => {
+    const purchasedSKUs = observable.set(new Set<string>())
+    const products = observable.box(Array<Product>())
+    const isFetchingProducts = observable.box(false)
+
+    purchaseUpdatedListener(flow(function * (purchase: Purchase) {
+      const receipt = purchase.transactionReceipt
+      if (receipt) {
+        // TODO: Validation
```

Comment:
How to perform validation properly in this case?

Reply:
FYR: https://github.com/dooboolab/react-native-iap#receipt-validation